akka-typed/src/main/scala/akka/typed/cluster/Cluster.scala

```diff
@@ -0,0 +1,256 @@
+          adaptedCluster.subscribe(subscriber.toUntyped, initialStateMode = ClusterEvent.initialStateAsEvents, eventClass)
+          Actor.same
+
+        case Unsubscribe(subscriber) ⇒
+          adaptedCluster.unsubscribe(subscriber.toUntyped)
+          Actor.same
+
+        case OnSelfUp(subscriber) ⇒
+          if (cluster.selfMember.status == MemberStatus.Up) subscriber ! SelfUp(adaptedCluster.state)
+          else {
+            ctx.watch(subscriber)
+            upSubscribers = subscriber :: upSubscribers
+          }
+          Actor.same
+
+        case GetCurrentState(sender) ⇒
+          sender ! adaptedCluster.state
+          Actor.same
+
+        case ClusterEvent.MemberUp(member) if member.uniqueAddress == cluster.selfMember.uniqueAddress ⇒
+          upSubscribers.foreach(_ ! SelfUp(adaptedCluster.state))
+          upSubscribers = Nil
+          Actor.same
+
+        case _: ClusterEvent.MemberUp ⇒
+          Actor.same
+
+      }
+    }.onSignal {
+
+      case (_, Terminated(ref)) ⇒
+        upSubscribers = upSubscribers.filterNot(_ == ref)
+        Actor.same
+
+    }.narrow[ClusterStateSubscription]
+  }
+
+  private def managerBehavior(adaptedCluster: akka.cluster.Cluster) = Actor.immutable[ClusterCommand]((ctx, msg) ⇒
+    msg match {
+      case Join(address) ⇒
+        adaptedCluster.join(address)
+        Actor.same
+
+      case Leave(address) ⇒
+        adaptedCluster.leave(address)
+        Actor.same
+
+      case Down(address) ⇒
+        adaptedCluster.down(address)
+        Actor.same
+
+      case JoinSeedNodes(addresses) ⇒
+        adaptedCluster.joinSeedNodes(addresses)
+        Actor.same
+
+    }
+
+  )
+
+}
+
+/**
+ * INTERNAL API:
+ */
+@InternalApi
+private[akka] final class AdapterClusterImpl(system: ActorSystem[_]) extends Cluster {
+  import AdapterClusterImpl._
+
+  require(system.isInstanceOf[ActorSystemAdapter[_]], "only adapted actor systems can be used for cluster features")
+  private val untypedSystem = ActorSystemAdapter.toUntyped(system)
+  private def extendedUntyped = untypedSystem.asInstanceOf[ExtendedActorSystem]
+  private val adaptedCluster = akka.cluster.Cluster(untypedSystem)
+
+  def selfMember =
+    adaptedCluster.state.members.find(_.uniqueAddress == adaptedCluster.selfUniqueAddress).getOrElse(
+      // Not sure if this fake-it-until-you-make it strategy makes sense, but wanted to avoid using an option here
+      new Member(adaptedCluster.selfUniqueAddress, 0, MemberStatus.Joining, adaptedCluster.selfRoles)
+    )
+  def isTerminated = adaptedCluster.isTerminated
+  def state = adaptedCluster.state
+
+  override lazy val subscriptions: ActorRef[ClusterStateSubscription] = extendedUntyped.systemActorOf(
+    PropsAdapter(subscriptionsBehavior(adaptedCluster)), "cluster-state-subscriptions")
+  override lazy val manager: ActorRef[ClusterCommand] = extendedUntyped.systemActorOf(
+    PropsAdapter(managerBehavior(adaptedCluster)), "cluster-command-manager")
```

Comment:
we probably have a mix of conventions for actor names, but I think most cluster actors are with camel case, lower first